### PR TITLE
Add RabbitFeed.route_prefix_extension

### PR DESCRIPTION
**Pull request by PeterWuMC**
@joshuafleck @calo81 @daniel-barlow 

This will append the `route_prefix_extension` to the routing key in producer and consumer

```ruby 
# in rabbit_feed.yml
route_prefix_extension: 'foobar'
#=> routing key will be: env.foobar.application.event

# without the route_prefix_extension
#=> routing key will be: env.application.event
```

This is my another attempt to make it simpler for testing in integration environment.

In the rails integration env, we can set that `route_prefix_extension` with deploy user name

what do you think @joshuafleck

### CODE REVIEW

**Comment by joshuafleck** [comment]
Looks pretty cool! This could allow us to build one shared integration rabbitmq cluster, which would save us some ££. Would like to see a demo of CMM using this in action on integration, but happy with the approach :+1: 

signed-off-by: @joshuafleck

**Comment by PeterWuMC** [comment]
> Would like to see a demo of CMM using this in action on integration, but happy with the approach

so, i sort of got it working:
cmm changes: https://github.com/simplybusiness/call_me_maybe/compare/test-pwu?expand=1
chopin changes: https://github.com/simplybusiness/chopin/compare/test-pwu?expand=1

not exactly the way i would imagine it to be, however call me maybe uses pooka and it does not load up cmm rails environment, hence need to brute force the prefix in

**Comment by joshuafleck** [comment]
One issue: if you've got multiple people running their own consumers, the consumers would all consume from the same queue (so far, you've only changed how the messages are routed from the exchange to the queue). This means that the messages would go to 1 of the N consumers. You'd also need to add the prefix to the queue name and ensure that the consumer consumes from the queue with the prefixed name.

File: lib/rabbit_feed.rb [hunk 1]
```diff
@@ -49,6 +50,10 @@ def default_logger
     end
   end
 
+  def route_prefix_extension
+    ".#{@route_prefix_extension}" if !!@route_prefix_extension && !@route_prefix_extension.empty?
```
Comment (joshuafleck) [inline lib/rabbit_feed.rb]: You can shorten this to `unless @route_prefix_extension.blank?`

```
irb(main):001:0> require 'rabbit_feed'
=> true
irb(main):002:0> nil.blank?
=> true
irb(main):003:0> ''.blank?
=> true
irb(main):004:0> 'test'.blank?
=> false
```

File: lib/rabbit_feed/client.rb [hunk 2]
```diff
@@ -160,6 +161,10 @@ def parse_options argv
           opts[:environment] = arg
         end
 
+        o.on '-x', '--prefix VAL', 'Prefix for rabbit queue routing key' do |arg|
```
Comment (joshuafleck) [inline lib/rabbit_feed/client.rb]: How about: `Prefix to append to the routing and queue naming allowing exclusive publishing/consuming.`? Open to suggestions here

File: lib/rabbit_feed.rb [hunk 3]
```diff
@@ -49,6 +50,10 @@ def default_logger
     end
   end
 
+  def route_prefix_extension
```
Comment (joshuafleck) [inline lib/rabbit_feed.rb]: Was there a compelling reason to define this in the `RabbitFeed` module? I ask because I think that moving this parameter to the configuration file might make this parameter easier to set and manage per environment. Also, this might prevent the need for the extra command line parameter as Pooka also reads from `rabbit_feed.yml`. Apologies for the last minute comment, but this just came to mind.

File: lib/rabbit_feed/client.rb [hunk 4]
```diff
@@ -160,6 +161,10 @@ def parse_options argv
           opts[:environment] = arg
         end
 
+        o.on '-x', '--prefix VAL', 'Prefix to append to the routing and queue naming allowing exclusive publishing/consuming' do |arg|
```
Comment (joshuafleck) [inline lib/rabbit_feed/client.rb]: Thinking about it further, I don't like the idea of exposing this as a command line argument as this is config-related and not related to the actual running of the client application.

Comment (PeterWuMC) [inline lib/rabbit_feed/client.rb]: but you have `application` in there, which uses in queue / routing as well. nevertheless, i am looking to remove it

File: spec/lib/rabbit_feed/producer_spec.rb [hunk 5]
```diff
@@ -68,6 +68,20 @@ module RabbitFeed
           subject
         end
       end
+
+      context 'when RabbitFeed.route_prefix_extension is set' do
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/producer_spec.rb]: `RabbitFeed.route_prefix_extension` no longer exists - could change to `when a route_prefix_extension is set`

File: spec/lib/rabbit_feed/consumer_connection_spec.rb [hunk 6]
```diff
@@ -29,6 +29,18 @@ module RabbitFeed
         expect(bunny_channel).to receive(:prefetch).with(1)
         subject
       end
+
+      context 'when RabbitFeed.route_prefix_extension is set' do
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/consumer_connection_spec.rb]: `RabbitFeed.route_prefix_extension` no longer exists - could change to `when a route_prefix_extension is set`

File: spec/lib/rabbit_feed/configuration_spec.rb [hunk 7]
```diff
@@ -11,6 +11,18 @@ module RabbitFeed
       subject { (described_class.new options).queue }
 
       it { should eq 'test.rabbit_feed' }
+
+      context 'when RabbitFeed.route_prefix_extension is set' do
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/configuration_spec.rb]: `RabbitFeed.route_prefix_extension` no longer exists - could change to `when a route_prefix_extension is set`

**Comment by joshuafleck** [comment]
One instance you missed, plus still need to update the version number. Also, could you please add this option to the config table in the readme: https://github.com/simplybusiness/rabbit_feed#configuration

**Comment by joshuafleck** [comment]
Should run the `./run_example` script to pick up the version bump in the example projects, but looks great otherwise! :+1: 

signed-off-by: @joshuafleck
